python/cudf/cudf/core/frame.py

```diff
@@ -489,6 +489,9 @@ def _hash_partition(
         )
         output = self.__class__._from_table(output)
         output._copy_categories(self, include_index=keep_index)
+        # Convert empty offsets to list of zeros.
+        # This is needed when `self` is an empty Frame.
+        offsets = offsets or [0] * num_partitions
```

Comment:
Could we instead push this empty dataframe handling into cython `libcudf.hash.hash_partition` function?

Reply:
Sure - I'll try to push it down